internal/scaffold/olm-catalog/descriptor/parse.go

```diff
@@ -186,27 +186,55 @@ func parseDescription(comments []string) string {
 	return strings.Join(lines, " ")
 }
 
-var jsonTagRe = regexp.MustCompile(`json:"([a-zA-Z0-9,]+|-)"`)
-
 const (
 	inlinedTag = "##inline##"
 	ignoredTag = "##ignore##"
 )
 
-func parsePathFromJSONTags(tags string) (string, error) {
-	tagMatches := jsonTagRe.FindStringSubmatch(tags)
-	if len(tagMatches) > 1 {
-		ts := strings.Split(tagMatches[1], ",")
+// getPathFromMember constructs a path from data in member, either from
+// its struct tags or name.
+func getPathFromMember(member types.Member) (string, error) {
+	// Unexported fields should be ignored in downstream processing.
+	if isNotExported(member.Name) {
+		return ignoredTag, nil
+	}
+	tags, err := structtag.Parse(member.Tags)
+	if err != nil {
+		return "", err
+	}
+	jsonTag, err := tags.Get("json")
+	if err == nil {
+		// Parse returns an error if no JSON tag is in tags, at which point we'll
+		// use another method to get path.
 		switch {
-		case len(ts) == 2 && ts[1] == "inline":
+		case contains(jsonTag.Options, "inline"):
 			return inlinedTag, nil
-		case len(ts) == 1 && ts[0] == "-":
+		case jsonTag.Name == "-" || contains(jsonTag.Options, "-"):
```

Comment:
I read over https://golang.org/pkg/encoding/json/#Marshal and realized there's the special case of the actually using the key name `-`, by defining `json:"-,"` so I think we need:
```go
case jsonTag.Name == "-" && len(jsonTag.Options) > 0:
	return jsonTag.Name
```
I'm not sure if that's actually valid in the path string though... :thinking: 

Also, what's the case for ignoring `contains(jsonTag.Options, "-")`? I don't see anything in the docs about using `-` as an option (e.g. `json:"myField,-"`) to ignore the field.

Reply:
Hmm I thought I had read somewhere that `json:"myField,-"` is valid, but since I can't find what I read I'll remove that condition and add the one you suggested.

Reply:
It does seem odd, but from a JSON-only perspective, something like this is valid:
```json
{
    "-": "my weirdly named field"
}

Like I said, not sure what the thing parsing this path from the CSV would do. I would think it would either work no problem (in which case we should allow `-` in the path) or it would fail (in which case we should probably return an error here).